### PR TITLE
Adds replace option for package publishing f/@starbelly

### PR DESCRIPTION
**Pull request by supersimple**
Adds support for replacing an existing release

### CODE REVIEW

File: src/hex_api_release.erl [hunk 1]
```diff
@@ -79,6 +80,41 @@ publish(Config, Tarball) when is_map(Config) and is_binary(Tarball) ->
...
+%%      <<"publisher">> => nil,<<"requirements">> => #{},
+%%      <<"retirement">> => nil,
+%%      <<"updated_at">> => <<"2019-07-28T21:12:11Z">>,
+%%      <<"url">> => <<"https://hex.pm/api/packages/package/releases/1.0.0">>,
+%%      <<"version">> => <<"1.0.0">>
+%%      }}}
+%% '''
+%% @end
+-spec replace(hex_core:config(), binary()) -> hex_api:response().
+replace(Config, Tarball) when is_map(Config) and is_binary(Tarball) ->
```
Comment (wojtekmach) [inline src/hex_api_release.erl]: so my initial though we'd have `publish(Config, Tarball, [{replace, true}])` (and in the future other potential options) but I think making this a separate explicit operation is a good idea too. Do you have a preference @ericmj?

Comment (ericmj) [reply src/hex_api_release.erl]: I think an option on `publish` is preferable. But regardless of this we need `publish` to default to `?replace=false`.

File: src/hex_api_release.erl [hunk 2]
```diff
@@ -72,12 +75,46 @@ get(Config, Name, Version) when is_map(Config) and is_binary(Name) and is_binary
 %% '''
 %% @end
 -spec publish(hex_core:config(), binary()) -> hex_api:response().
-publish(Config, Tarball) when is_map(Config) and is_binary(Tarball) ->
-    Path = hex_api:build_repository_path(Config, ["publish"]),
+publish(Config, Tarball) -> publish(Config, Tarball, []).
```
Comment (supersimple) [inline src/hex_api_release.erl]: @wojtekmach @ericmj @starbelly how would i default this replace arg to false?

File: src/hex_api_release.erl [hunk 3]
```diff
@@ -72,12 +75,46 @@ get(Config, Name, Version) when is_map(Config) and is_binary(Name) and is_binary
...
+%%      <<"retirement">> => nil,
+%%      <<"updated_at">> => <<"2019-07-28T21:12:11Z">>,
+%%      <<"url">> => <<"https://hex.pm/api/packages/package/releases/1.0.0">>,
+%%      <<"version">> => <<"1.0.0">>
+%%      }}}
+%% '''
+%% @end
+-spec publish(hexcore:config(), binary(), publish_params()) -> hex_api:response().
+publish(Config, Tarball, Params) when is_map(Config) andalso is_binary(Tarball) andalso is_list(Params)->
+    QueryString = hex_api:encode_query_string(Params),
```
Comment (ericmj) [inline src/hex_api_release.erl]: @supersimple You can do this to set the default replace value:

```suggestion
    QueryString = hex_api:encode_query_string([{replace, proplists:get_value(replace, params, false)}]),
```

File: src/hex_api_release.erl [hunk 4]
```diff
@@ -72,12 +75,46 @@ get(Config, Name, Version) when is_map(Config) and is_binary(Name) and is_binary
...
+%%      <<"retirement">> => nil,
+%%      <<"updated_at">> => <<"2019-07-28T21:12:11Z">>,
+%%      <<"url">> => <<"https://hex.pm/api/packages/package/releases/1.0.0">>,
+%%      <<"version">> => <<"1.0.0">>
+%%      }}}
+%% '''
+%% @end
+-spec publish(hexcore:config(), binary(), publish_params()) -> hex_api:response().
+publish(Config, Tarball, Params) when is_map(Config) andalso is_binary(Tarball) andalso is_list(Params)->
+    QueryString = hex_api:encode_query_string([{replace, proplists:get_value(replace, params, false)}]),
```
Comment (ericmj) [inline src/hex_api_release.erl]: Oops.

```suggestion
    QueryString = hex_api:encode_query_string([{replace, proplists:get_value(replace, Params, false)}]),
```

**Comment by ericmj** [comment]
Thank you! 💛

**Pull request merged**